create_report_rmd.R

```diff
@@ -0,0 +1,124 @@
+  if( !is.na(dict$Subquestion[ idx[1] ]) ){
+    # remove square brackets at beginning and end, then rename columns:
+    colnames( responses ) <- str_extract( dict$Subquestion[idx], '(?<=\\[)[^{}]+(?=\\])')
+  } else{
+    colnames( responses ) <- "Response"
+  }
+  return( responses )
+}
+
+idx <- which( is.na( dict$`Question number` )) 
+          # which( grepl('upload', dict$`Type of variable`)),
+          # which ( grepl('filecount', dict$`Question number`) ))
+all_questions <- dict$`Question number`[-idx]
+idx <- which( all_questions == "1" )
+all_questions <- unique( all_questions[ idx:length( all_questions )] )
+n_questions <- length( all_questions )
+    
+write_lines("---
+title: \"Arboviral Surveillance and Response Capacity Survey 2021\"
+output: html_document
+params:
+  country: \"Afghanistan\"
+---
+
+```{r setup, include=FALSE, echo = FALSE}
+# Basic knitr options
+library(knitr)
+opts_chunk$set(comment = NA, 
+               echo = FALSE,
+               warning = FALSE, 
+               message = FALSE, 
+               error = TRUE, 
+               cache = FALSE,
+               fig.width = 9.64,
+               fig.height = 5.9,
+               fig.path = 'figures/')
+options(scipen = '999')
+```
+
+```{r}
+
+library(tidyverse)
+library(stringr)
+library(pander)
+library(knitr)
+library(flextable)
+library(lubridate)
+
+if( file.exists(\"data/data.RData\")){
+  load(\"data/data.RData\")
+} else{
+  message(\"You need the data!\")
+}
+
+data <- data %>% filter( SI01 == params$country )
+
+if( file.exists(\"misc/Data_dictionary_Survey375147.csv\")){
+  dict <- read_csv(\"misc/Data_dictionary_Survey375147.csv\")
+} else{
+  message(\"You need the data dictionary!\")
+}
+
+# utility function 
+get_responses <- function( question_number ){
+  idx <- which( dict$`Question number` == question_number )
+  var_names <- dict$`Variable name`[idx]
+  responses <- data %>% select( all_of(var_names) )
+  if( !is.na(dict$Subquestion[ idx[1] ]) ){
+    # remove square brackets at beginning and end, then rename columns:
+    colnames( responses ) <- str_extract( dict$Subquestion[idx], '(?<=\\\\[)[^{}]+(?=\\\\])')
+  } else{
+    colnames( responses ) <- \"Response\"
+  }
+  return( responses )
+}
+```
+", "report.Rmd")
+    
+for( j in 1:n_questions ){
+  num <- all_questions[j]
+  idx <- which( dict$`Question number` == num)
+  question <- str_trim( dict$Description[idx[1]] )
+  write_lines(paste0("\n#### ",num,". ",question,"\n"), "report.Rmd", append=TRUE)
+  # response <- get_responses( df, num )
+  # sink(outfile, append=TRUE)
+  # cat( pander(response), sep="\n" )
+  # sink()
+  if( length(idx)==1){
+    var_name <- dict$`Variable name`[idx]
+    write_lines(paste0("`r data$",var_name,"` \n\n*****"), "report.Rmd", append=TRUE)
+  } else{
+    write_lines(paste0("```{r}\nresponses <- get_responses(\"",num,"\")\nknitr::kable(responses)\n```\n\n*****"),"report.Rmd",append=TRUE)
+  }
+}
```

Comment:
After running this, I knit the `report.Rmd` manually. Looks great. Awesome formatting.

Reply:
see comments above... this is work in progress!